gcloud-java-compute/src/main/java/com/google/gcloud/compute/RawImageConfiguration.java

```diff
@@ -0,0 +1,217 @@
+      this.source = imagePb.getRawDisk().getSource();
+      this.archiveSizeBytes = imagePb.getArchiveSizeBytes();
+    }
+
+    /**
+     * Sets the format used to encode and transmit the block device. Supported value is only
+     * {@code TAR}. This is just a container and transmission format, not a runtime format.
+     */
+    public Builder containerType(ContainerType containerType) {
+      this.containerType = containerType;
+      return this;
+    }
+
+    /**
+     * Sets the SHA1 checksum of the disk image before unpackaging.
+     */
+    public Builder sha1Checksum(String sha1Checksum) {
+      this.sha1Checksum = sha1Checksum;
+      return this;
+    }
+
+    /**
+     * Sets the full Google Cloud Storage URL where the disk image is stored (e.g.
+     * {@code gs://bucket/file}).
+     */
+    public Builder source(String source) {
+      this.source = source;
+      return this;
+    }
+
+    Builder archiveSizeBytes(Long archiveSizeBytes) {
+      this.archiveSizeBytes = archiveSizeBytes;
+      return this;
+    }
+
+    /**
+     * Creates a {@code RawImageConfiguration} object.
+     */
+    @Override
+    public RawImageConfiguration build() {
+      return new RawImageConfiguration(this);
+    }
+  }
+
+  private RawImageConfiguration(Builder builder) {
+    super(builder);
+    this.source = checkNotNull(builder.source);
+    this.containerType = builder.containerType;
+    this.sha1Checksum = builder.sha1Checksum;
+    this.archiveSizeBytes = builder.archiveSizeBytes;
+  }
+
+  /**
+   * Returns the format used to encode and transmit the block device. This is just a container and
+   * transmission format, not a runtime format.
+   */
+  public ContainerType containerType() {
+    return containerType;
+  }
+
+  /**
+   * Returns the SHA1 checksum of the disk image before unpackaging.
+   */
+  public String sha1Checksum() {
+    return sha1Checksum;
+  }
+
+  /**
+   * Returns the full Google Cloud Storage URL where the disk image is stored (e.g.
+   * {@code gs://bucket/file}).
+   */
+  public String source() {
+    return source;
+  }
+
+  /**
+   * Returns the size of the image archive stored in Google Cloud Storage (in bytes).
+   */
+  public Long archiveSizeBytes() {
+    return archiveSizeBytes;
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  MoreObjects.ToStringHelper toStringHelper() {
+    return super.toStringHelper()
+        .add("source", source)
+        .add("containerType", containerType)
+        .add("sha1Checksum", sha1Checksum)
+        .add("archiveSizeBytes", archiveSizeBytes);
+  }
+
+  @Override
+  public final int hashCode() {
+    return Objects.hash(baseHashCode(), source, containerType, sha1Checksum, archiveSizeBytes);
+  }
+
+  @Override
+  public final boolean equals(Object obj) {
+    return obj instanceof RawImageConfiguration && baseEquals((RawImageConfiguration) obj);
```

Comment:
I think Ajay has a point. we should probably make the `baseEquals` check `getClass()` equality just in case that the toPb() equality (based on map equality) is true (if all values are the same). Because the subclasses may have their own specific *required* fields that many not be an issue now but I see no reason why we should make it work regardless.